core-rust/state-manager/src/receipt.rs

```diff
@@ -85,63 +94,68 @@ impl From<TransactionOutcome> for LedgerTransactionOutcome {
                         .collect(),
                 )
             }
-            TransactionOutcome::Failure(error) => LedgerTransactionOutcome::Failure(error),
+            TransactionOutcome::Failure(error) => Self::Failure(error),
         }
     }
 }
 
+#[derive(Debug, Clone, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
+pub struct LocalTransactionReceipt {
+    pub on_ledger: LedgerTransactionReceipt,
+    pub local_execution: LocalTransactionExecution,
+}
+
 #[derive(Debug, Clone, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
 pub struct LedgerTransactionReceipt {
     pub outcome: LedgerTransactionOutcome,
+    pub substate_changes: SubstateChanges,
+}
+
+#[derive(Debug, Clone, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
+pub struct LocalTransactionExecution {
+    pub outcome: DetailedTransactionOutcome,
     // The breakdown of the fee
     pub fee_summary: FeeSummary,
     // Which vault/s paid the fee
     pub fee_payments: IndexMap<ObjectId, Decimal>,
     pub application_logs: Vec<(Level, String)>,
-    pub substate_changes: SubstateChanges,
     pub state_update_summary: StateUpdateSummary,
     // These will be removed once we have the parent_map for the toolkit to use
     pub resource_changes: IndexMap<usize, Vec<ResourceChange>>,
     pub next_epoch: Option<(BTreeMap<ComponentAddress, Validator>, u64)>,
 }
 
 impl LedgerTransactionReceipt {
-    pub fn get_hash(&self) -> LedgerReceiptHash {
-        LedgerReceiptHash::for_receipt(self)
+    pub fn get_consensus_receipt(&self) -> ConsensusReceipt {
```

Comment:
If lifetime is quite big, and the consensus receipt is a transient structure, we could also consider it containing a reference or Cow (with a lifetime) to outcome (using a Cow rather than a lifetime would also enable it to own the outcome so it can be a deserialisation target as well).